azurerm/internal/services/sentinel/sentinel_automation_rule_resource.go

```diff
@@ -0,0 +1,661 @@
+
+		actionIncident, actionPlaybook := flattenAutomationRuleActions(prop.Actions)
+
+		if err := d.Set("action_incident", actionIncident); err != nil {
+			return fmt.Errorf("setting `action_incident`: %v", err)
+		}
+		if err := d.Set("action_playbook", actionPlaybook); err != nil {
+			return fmt.Errorf("setting `action_playbook`: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceSentinelAutomationRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sentinel.AutomationRulesClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.AutomationRuleID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Delete(ctx, id.ResourceGroup, OperationalInsightsResourceProvider, id.WorkspaceName, id.Name)
+	if err != nil {
+		return fmt.Errorf("deleting %s: %+v", id, err)
+	}
+
+	return nil
+}
+
+func expandAutomationRuleConditions(input []interface{}) *[]securityinsight.BasicAutomationRuleCondition {
+	if len(input) == 0 {
+		return nil
+	}
+
+	out := make([]securityinsight.BasicAutomationRuleCondition, 0, len(input))
+	for _, b := range input {
+		b := b.(map[string]interface{})
+
+		out = append(out, &securityinsight.AutomationRulePropertyValuesCondition{
+			ConditionProperties: &securityinsight.AutomationRulePropertyValuesConditionConditionProperties{
+				PropertyName:   securityinsight.AutomationRulePropertyConditionSupportedProperty(b["property"].(string)),
+				Operator:       securityinsight.AutomationRulePropertyConditionSupportedOperator(b["operator"].(string)),
+				PropertyValues: utils.ExpandStringSlice(b["values"].([]interface{})),
+			},
+			ConditionType: securityinsight.ConditionTypeProperty,
+		})
+	}
+	return &out
+}
+
+func flattenAutomationRuleConditions(conditions *[]securityinsight.BasicAutomationRuleCondition) interface{} {
+	if conditions == nil {
+		return nil
+	}
+
+	out := make([]interface{}, 0, len(*conditions))
+	for _, condition := range *conditions {
+		condition := condition.(securityinsight.AutomationRulePropertyValuesCondition)
+
+		var (
+			property string
+			operator string
+			values   []interface{}
+		)
+		if p := condition.ConditionProperties; p != nil {
+			property = string(p.PropertyName)
+			operator = string(p.Operator)
+			values = utils.FlattenStringSlice(p.PropertyValues)
+		}
+
+		out = append(out, map[string]interface{}{
+			"property": property,
+			"operator": operator,
+			"values":   values,
+		})
+	}
+	return out
+}
+
+func expandAutomationRuleActions(d *schema.ResourceData, defaultTenantId string) (*[]securityinsight.BasicAutomationRuleAction, error) {
+	actionIncident, err := expandAutomationRuleActionIncident(d.Get("action_incident").([]interface{}))
+	if err != nil {
+		return nil, err
+	}
+	actionPlaybook := expandAutomationRuleActionPlaybook(d.Get("action_playbook").([]interface{}), defaultTenantId)
+
+	if len(actionIncident)+len(actionPlaybook) == 0 {
+		return nil, nil
+	}
+
```

Comment:
```suggestion
	if len(actionIncident)+len(actionPlaybook) >= 2 { //For any more than one action we need to explictly put together the out object.
		actionIncidentSize := cap(actionIncident)
		actionPlaybookSize := cap(actionPlaybook)
		totalSize := actionIncidentSize+actionPlaybookSize //We expect this to be the amount of either incident actions or playbook actions
		out := make([]securityinsight.BasicAutomationRuleAction, len(actionIncident)+len(actionPlaybook))
		copy(out[0:actionIncidentSize], actionIncident[:]) //Start at zero, add on incident actions
		copy(out[actionIncidentSize:totalSize], actionPlaybook[:]) //Patch on playbook actions
		return &out, nil
	}
```
This should fix the issue I mentioned, have tested myself with both and one playbook with two incident actions, then did some extra ones (excuse if its sloppy, never actually written golang before):
![image](https://user-images.githubusercontent.com/29356754/116608524-33a3e900-a92b-11eb-9575-2b97f8437573.png)